### PR TITLE
Fix the build of superlu_dist for openEuler

### PR DESCRIPTION
**Pull request by martin-g**
Successful OBS builds: https://obs.openhpc.community/package/show/home:mgrigorov/superlu_dist-gnu12-openmpi4

There is a test mapping already!

### CODE REVIEW

**Comment by martin-g** [comment]
The tests fail with:
```
2023-02-28T17:59:23.5965806Z not ok 1 [libs/superLU_dist] PDGSSVX with full (default) options (slurm/gnu12/openmpi4)
2023-02-28T17:59:23.5966128Z # (from function `run_mpi_binary' in file ./common/functions, line 392,
2023-02-28T17:59:23.5966260Z #  in test file rm_execution, line 26)
2023-02-28T17:59:23.5966532Z #   `run_mpi_binary ./pddrive "-r 2 -c 2 g20.rua" $NODES $TASKS' failed
2023-02-28T17:59:23.5966655Z # job script = /tmp/job.ohpc.13925
2023-02-28T17:59:23.5966759Z # Batch job 6 submitted
2023-02-28T17:59:23.5966822Z #  
2023-02-28T17:59:23.5966915Z # Job 6 failed...
2023-02-28T17:59:23.5967025Z # Reason=NonZeroExitCode
2023-02-28T17:59:23.5967103Z #  
2023-02-28T17:59:23.5967246Z # [prun] Master compute host = 9cdf6c589f49
2023-02-28T17:59:23.5967370Z # [prun] Resource manager = slurm
2023-02-28T17:59:23.5967655Z # [prun] Launch cmd = mpirun ./pddrive -r 2 -c 2 g20.rua (family=openmpi4)
2023-02-28T17:59:23.5968200Z # Number of processes 2 is smaller than NPROW * NPCOL 4Number of processes 2 is smaller than NPROW * NPCOL 4--------------------------------------------------------------------------
2023-02-28T17:59:23.5968374Z # Primary job  terminated normally, but 1 process returned
2023-02-28T17:59:23.5968651Z # a non-zero exit code. Per user-direction, the job has been aborted.
2023-02-28T17:59:23.5968948Z # --------------------------------------------------------------------------
2023-02-28T17:59:23.5969235Z # --------------------------------------------------------------------------
2023-02-28T17:59:23.5969551Z # mpirun detected that one or more processes exited with non-zero status, thus causing
2023-02-28T17:59:23.5969937Z # the job to be terminated. The first process to do so was:
2023-02-28T17:59:23.5970023Z # 
2023-02-28T17:59:23.5970123Z #   Process name: [[43376,1],0]
2023-02-28T17:59:23.5970221Z #   Exit code:    255
2023-02-28T17:59:23.5970521Z # --------------------------------------------------------------------------
```

**Comment by adrianreber** [comment]
Maybe the test needs to be adapted to the smaller amount of CPUs in GitHub Actions

**Comment by martin-g** [comment]
I was able to get more context of the error after changing `TASKS` to [4](https://github.com/openhpc/ohpc/pull/1721/files#diff-b9935557fe9878a1848f237611c88dd9344e389e5bff77a86daf76620469dff0R17). It does not print this error when its value is `2` (due to SIMPLE_CI) or `1`:

```
2023-03-01T14:35:35.8641440Z FAIL libs/superlu_dist/ohpc-tests/test_mpi_families (exit status: 1)
2023-03-01T14:35:35.8641539Z ================================================
2023-03-01T14:35:35.8641790Z Log file: ./tests/libs/superlu_dist/tests/family-gnu12-openmpi4/rm_execution.log
2023-03-01T14:35:35.8641861Z 1..1
2023-03-01T14:35:35.8642044Z not ok 1 [libs/superLU_dist] PDGSSVX with full (default) options (slurm/gnu12/openmpi4)
2023-03-01T14:35:35.8642277Z # (from function `run_mpi_binary' in file ./common/functions, line 392,
2023-03-01T14:35:35.8642386Z #  in test file rm_execution, line 26)
2023-03-01T14:35:35.8642613Z #   `run_mpi_binary ./pddrive "-r 2 -c 2 g20.rua" $NODES $TASKS' failed
2023-03-01T14:35:35.8642757Z # job script = /tmp/job.ohpc.5413
2023-03-01T14:35:35.8642835Z # Batch job 6 submitted
2023-03-01T14:35:35.8642909Z #  
2023-03-01T14:35:35.8642993Z # Job 6 failed...
2023-03-01T14:35:35.8643092Z # Reason=NonZeroExitCode
2023-03-01T14:35:35.8643162Z #  
2023-03-01T14:35:35.8643283Z # [prun] Master compute host = 828159c1e482
2023-03-01T14:35:35.8643386Z # [prun] Resource manager = slurm
2023-03-01T14:35:35.8643616Z # [prun] Launch cmd = mpirun ./pddrive -r 2 -c 2 g20.rua (family=openmpi4)
2023-03-01T14:35:35.8643828Z # [828159c1e482:9432 :0:9432] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x208cb60)
2023-03-01T14:35:35.8643927Z # ==== backtrace (tid:   9432) ====
2023-03-01T14:35:35.8644215Z #  0  /opt/ohpc/pub/mpi/ucx-ohpc/1.11.2/lib/libucs.so.0(ucs_handle_error+0x254) [0x7fe2f41ddc34]
2023-03-01T14:35:35.8644468Z #  1  /opt/ohpc/pub/mpi/ucx-ohpc/1.11.2/lib/libucs.so.0(+0x27def) [0x7fe2f41dddef]
2023-03-01T14:35:35.8644714Z #  2  /opt/ohpc/pub/mpi/ucx-ohpc/1.11.2/lib/libucs.so.0(+0x280b6) [0x7fe2f41de0b6]
2023-03-01T14:35:35.8645072Z #  3  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/openmpi/mca_btl_vader.so(mca_btl_vader_poll_handle_frag+0x154) [0x7fe2f449ebc4]
2023-03-01T14:35:35.8645364Z #  4  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/openmpi/mca_btl_vader.so(+0x4e4c) [0x7fe2f449ee4c]
2023-03-01T14:35:35.8645733Z #  5  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libopen-pal.so.40(opal_progress+0x2c) [0x7fe30746838c]
2023-03-01T14:35:35.8646051Z #  6  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libopen-pal.so.40(ompi_sync_wait_mt+0x10d) [0x7fe30746ea9d]
2023-03-01T14:35:35.8646349Z #  7  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(ompi_comm_nextcid+0x169) [0x7fe307eb57e9]
2023-03-01T14:35:35.8646640Z #  8  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(ompi_comm_create+0x286) [0x7fe307eaea66]
2023-03-01T14:35:35.8646931Z #  9  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(MPI_Comm_create+0x74) [0x7fe307eea734]
2023-03-01T14:35:35.8647151Z # 10  /opt/ohpc/pub/libs/gnu12/openmpi4/superlu_dist/6.4.0/lib/libsuperlu_dist.so.5(superlu_gridmap+0x107) [0x7fe309fdaf67]
2023-03-01T14:35:35.8647368Z # 11  /opt/ohpc/pub/libs/gnu12/openmpi4/superlu_dist/6.4.0/lib/libsuperlu_dist.so.5(superlu_gridinit+0x1ad) [0x7fe309fdb21d]
2023-03-01T14:35:35.8647465Z # 12  ./pddrive() [0x4023eb]
2023-03-01T14:35:35.8647585Z # 13  /usr/lib64/libc.so.6(+0x2d210) [0x7fe307ca0210]
2023-03-01T14:35:35.8647713Z # 14  /usr/lib64/libc.so.6(__libc_start_main+0x7c) [0x7fe307ca02bc]
2023-03-01T14:35:35.8647804Z # 15  ./pddrive() [0x402895]
2023-03-01T14:35:35.8647890Z # =================================
2023-03-01T14:35:35.8648015Z # [828159c1e482:09432] *** Process received signal ***
2023-03-01T14:35:35.8648139Z # [828159c1e482:09432] Signal: Segmentation fault (11)
2023-03-01T14:35:35.8648489Z # [828159c1e482:09432] Signal code:  (-6)
2023-03-01T14:35:35.8648614Z # [828159c1e482:09432] Failing at address: 0x3e8000024d8
2023-03-01T14:35:35.8648735Z # [828159c1e482:09432] [ 0] /usr/lib64/libc.so.6(+0x41070)[0x7fe307cb4070]
2023-03-01T14:35:35.8649202Z # [828159c1e482:09432] [ 1] /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/openmpi/mca_btl_vader.so(mca_btl_vader_poll_handle_frag+0x154)[0x7fe2f449ebc4]
2023-03-01T14:35:35.8649541Z # [828159c1e482:09432] [ 2] /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/openmpi/mca_btl_vader.so(+0x4e4c)[0x7fe2f449ee4c]
2023-03-01T14:35:35.8649896Z # [828159c1e482:09432] [ 3] /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libopen-pal.so.40(opal_progress+0x2c)[0x7fe30746838c]
2023-03-01T14:35:35.8650247Z # [828159c1e482:09432] [ 4] /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libopen-pal.so.40(ompi_sync_wait_mt+0x10d)[0x7fe30746ea9d]
2023-03-01T14:35:35.8650578Z # [828159c1e482:09432] [ 5] /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(ompi_comm_nextcid+0x169)[0x7fe307eb57e9]
2023-03-01T14:35:35.8650908Z # [828159c1e482:09432] [ 6] /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(ompi_comm_create+0x286)[0x7fe307eaea66]
2023-03-01T14:35:35.8651240Z # [828159c1e482:09432] [ 7] /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(MPI_Comm_create+0x74)[0x7fe307eea734]
2023-03-01T14:35:35.8651466Z # [828159c1e482:09432] [ 8] /opt/ohpc/pub/libs/gnu12/openmpi4/superlu_dist/6.4.0/lib/libsuperlu_dist.so.5(superlu_gridmap+0x107)[0x7fe309fdaf67]
2023-03-01T14:35:35.8651697Z # [828159c1e482:09432] [ 9] /opt/ohpc/pub/libs/gnu12/openmpi4/superlu_dist/6.4.0/lib/libsuperlu_dist.so.5(superlu_gridinit+0x1ad)[0x7fe309fdb21d]
2023-03-01T14:35:35.8651797Z # [828159c1e482:09432] [10] ./pddrive[0x4023eb]
2023-03-01T14:35:35.8651937Z # [828159c1e482:09432] [11] /usr/lib64/libc.so.6(+0x2d210)[0x7fe307ca0210]
2023-03-01T14:35:35.8652088Z # [828159c1e482:09432] [12] /usr/lib64/libc.so.6(__libc_start_main+0x7c)[0x7fe307ca02bc]
2023-03-01T14:35:35.8652201Z # [828159c1e482:09432] [13] ./pddrive[0x402895]
2023-03-01T14:35:35.8652316Z # [828159c1e482:09432] *** End of error message ***
2023-03-01T14:35:35.8652522Z # [828159c1e482:9430 :0:9430] Caught signal 11 (Segmentation fault: address not mapped to object at address 0x208cb60)
2023-03-01T14:35:35.8652624Z # ==== backtrace (tid:   9430) ====
2023-03-01T14:35:35.8652911Z #  0  /opt/ohpc/pub/mpi/ucx-ohpc/1.11.2/lib/libucs.so.0(ucs_handle_error+0x254) [0x7f4fc373cc34]
2023-03-01T14:35:35.8653219Z #  1  /opt/ohpc/pub/mpi/ucx-ohpc/1.11.2/lib/libucs.so.0(+0x27def) [0x7f4fc373cdef]
2023-03-01T14:35:35.8653464Z #  2  /opt/ohpc/pub/mpi/ucx-ohpc/1.11.2/lib/libucs.so.0(+0x280b6) [0x7f4fc373d0b6]
2023-03-01T14:35:35.8653820Z #  3  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/openmpi/mca_btl_vader.so(mca_btl_vader_poll_handle_frag+0x154) [0x7f4fc3df9bc4]
2023-03-01T14:35:35.8654107Z #  4  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/openmpi/mca_btl_vader.so(+0x4e4c) [0x7f4fc3df9e4c]
2023-03-01T14:35:35.8654414Z #  5  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libopen-pal.so.40(opal_progress+0x2c) [0x7f4fdadcf38c]
2023-03-01T14:35:35.8654731Z #  6  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libopen-pal.so.40(ompi_sync_wait_mt+0x10d) [0x7f4fdadd5a9d]
2023-03-01T14:35:35.8655031Z #  7  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(ompi_comm_nextcid+0x169) [0x7f4fdb81c7e9]
2023-03-01T14:35:35.8655328Z #  8  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(ompi_comm_create+0x286) [0x7f4fdb815a66]
2023-03-01T14:35:35.8655631Z #  9  /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(MPI_Comm_create+0x74) [0x7f4fdb851734]
2023-03-01T14:35:35.8655837Z # 10  /opt/ohpc/pub/libs/gnu12/openmpi4/superlu_dist/6.4.0/lib/libsuperlu_dist.so.5(superlu_gridmap+0x107) [0x7f4fdd941f67]
2023-03-01T14:35:35.8656057Z # 11  /opt/ohpc/pub/libs/gnu12/openmpi4/superlu_dist/6.4.0/lib/libsuperlu_dist.so.5(superlu_gridinit+0x1ad) [0x7f4fdd94221d]
2023-03-01T14:35:35.8656154Z # 12  ./pddrive() [0x4023eb]
2023-03-01T14:35:35.8656276Z # 13  /usr/lib64/libc.so.6(+0x2d210) [0x7f4fdb607210]
2023-03-01T14:35:35.8656419Z # 14  /usr/lib64/libc.so.6(__libc_start_main+0x7c) [0x7f4fdb6072bc]
2023-03-01T14:35:35.8656513Z # 15  ./pddrive() [0x402895]
2023-03-01T14:35:35.8656602Z # =================================
2023-03-01T14:35:35.8657141Z # [828159c1e482:09430] *** Process received signal ***
2023-03-01T14:35:35.8657276Z # [828159c1e482:09430] Signal: Segmentation fault (11)
2023-03-01T14:35:35.8657446Z # [828159c1e482:09430] Signal code:  (-6)
2023-03-01T14:35:35.8657577Z # [828159c1e482:09430] Failing at address: 0x3e8000024d6
2023-03-01T14:35:35.8657714Z # [828159c1e482:09430] [ 0] /usr/lib64/libc.so.6(+0x41070)[0x7f4fdb61b070]
2023-03-01T14:35:35.8658102Z # [828159c1e482:09430] [ 1] /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/openmpi/mca_btl_vader.so(mca_btl_vader_poll_handle_frag+0x154)[0x7f4fc3df9bc4]
2023-03-01T14:35:35.8658430Z # [828159c1e482:09430] [ 2] /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/openmpi/mca_btl_vader.so(+0x4e4c)[0x7f4fc3df9e4c]
2023-03-01T14:35:35.8658775Z # [828159c1e482:09430] [ 3] /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libopen-pal.so.40(opal_progress+0x2c)[0x7f4fdadcf38c]
2023-03-01T14:35:35.8659123Z # [828159c1e482:09430] [ 4] /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libopen-pal.so.40(ompi_sync_wait_mt+0x10d)[0x7f4fdadd5a9d]
2023-03-01T14:35:35.8659441Z # [828159c1e482:09430] [ 5] /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(ompi_comm_nextcid+0x169)[0x7f4fdb81c7e9]
2023-03-01T14:35:35.8659774Z # [828159c1e482:09430] [ 6] /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(ompi_comm_create+0x286)[0x7f4fdb815a66]
2023-03-01T14:35:35.8660101Z # [828159c1e482:09430] [ 7] /opt/ohpc/pub/mpi/openmpi4-gnu12/4.1.4/lib/libmpi.so.40(MPI_Comm_create+0x74)[0x7f4fdb851734]
2023-03-01T14:35:35.8660332Z # [828159c1e482:09430] [ 8] /opt/ohpc/pub/libs/gnu12/openmpi4/superlu_dist/6.4.0/lib/libsuperlu_dist.so.5(superlu_gridmap+0x107)[0x7f4fdd941f67]
2023-03-01T14:35:35.8660566Z # [828159c1e482:09430] [ 9] /opt/ohpc/pub/libs/gnu12/openmpi4/superlu_dist/6.4.0/lib/libsuperlu_dist.so.5(superlu_gridinit+0x1ad)[0x7f4fdd94221d]
2023-03-01T14:35:35.8660677Z # [828159c1e482:09430] [10] ./pddrive[0x4023eb]
2023-03-01T14:35:35.8660815Z # [828159c1e482:09430] [11] /usr/lib64/libc.so.6(+0x2d210)[0x7f4fdb607210]
2023-03-01T14:35:35.8660974Z # [828159c1e482:09430] [12] /usr/lib64/libc.so.6(__libc_start_main+0x7c)[0x7f4fdb6072bc]
2023-03-01T14:35:35.8661084Z # [828159c1e482:09430] [13] ./pddrive[0x402895]
2023-03-01T14:35:35.8661264Z # [828159c1e482:09430] *** End of error message ***
2023-03-01T14:35:35.8661528Z # --------------------------------------------------------------------------
2023-03-01T14:35:35.8661682Z # Primary job  terminated normally, but 1 process returned
2023-03-01T14:35:35.8661919Z # a non-zero exit code. Per user-direction, the job has been aborted.
2023-03-01T14:35:35.8662164Z # --------------------------------------------------------------------------
2023-03-01T14:35:35.8662413Z # --------------------------------------------------------------------------
2023-03-01T14:35:35.8662617Z # mpirun noticed that process rank 1 with PID 9432 on node c0 exited on signal 11 (Segmentation fault).
```

It looks like https://github.com/openhpc/ohpc/pull/1681#issuecomment-1429068875

**Comment by martin-g** [comment]
Adding `export OMPI_MCA_btl="self,tcp"` helped for most of the tests!
Now only one fails:
```
1..9
2023-03-01T18:34:53.3795315Z ok 1 [libs/superLU_dist] PDGSSVX with full (default) options (slurm/gnu12/mpich)
2023-03-01T18:34:53.3795502Z ok 2 [libs/superLU_dist] pdgssvx_ABglobal with full (default) options (slurm/gnu12/mpich)
2023-03-01T18:34:53.3795636Z ok 3 [libs/superLU_dist] vary RHS (slurm/gnu12/mpich)
2023-03-01T18:34:53.3795771Z ok 4 [libs/superLU_dist] vary RHS ABglobal (slurm/gnu12/mpich)
2023-03-01T18:34:53.3796007Z ok 5 [libs/superLU_dist] re-use permutation vector (slurm/gnu12/mpich)
2023-03-01T18:34:53.3796263Z ok 6 [libs/superLU_dist] re-use permutation vector ABglobal (slurm/gnu12/mpich)
2023-03-01T18:34:53.3796507Z ok 7 [libs/superLU_dist] re-use symbolic factorization (slurm/gnu12/mpich)
2023-03-01T18:34:53.3796766Z ok 8 [libs/superLU_dist] re-use symbolic factorization ABglobal (slurm/gnu12/mpich)
2023-03-01T18:34:53.3796974Z not ok 9 [libs/superLU_dist] multi-grid (slurm/gnu12/mpich)
2023-03-01T18:34:53.3797206Z # (from function `run_mpi_binary' in file ./common/functions, line 381,
2023-03-01T18:34:53.3797315Z #  in test file rm_execution, line 99)
2023-03-01T18:34:53.3797503Z #   `run_mpi_binary ./pddrive4 "g20.rua" $NODES 10' failed
2023-03-01T18:34:53.3797610Z # job script = /tmp/job.ohpc.11907
2023-03-01T18:34:53.3797751Z # Batch job 23 submitted
2023-03-01T18:34:53.3797824Z #  
2023-03-01T18:34:53.3797948Z # cat: job.23.out: No such file or directory
2023-03-01T18:34:53.3798055Z FAIL rm_execution (exit status: 1)
```

**Comment by martin-g** [comment]
The PR is ready for review!
The last CI builds were green. I just squashed the commits into one.

File: tests/libs/superlu_dist/tests/rm_execution [hunk 1]
```diff
@@ -13,8 +13,11 @@ rm=$RESOURCE_MANAGER
 
 testname="libs/superLU_dist"
 
+if [ ! -z "$SIMPLE_CI" ]; then
+export OMPI_MCA_btl="self,tcp"
```
Comment (adrianreber) [inline tests/libs/superlu_dist/tests/rm_execution]: This line should be indented.

File: tests/libs/superlu_dist/tests/rm_execution [hunk 2]
```diff
@@ -13,8 +13,11 @@ rm=$RESOURCE_MANAGER
...
+fi
 NODES=2
-TASKS=`tasks_count 4`
+TASKS=4
```
Comment (adrianreber) [inline tests/libs/superlu_dist/tests/rm_execution:20]: Why is this change necessary? I thought `tasks_count` it the function which transparently tries to do the right thing if running in GitHub Actions?

**Review by adrianreber — CHANGES_REQUESTED** [review]

File: tests/libs/superlu_dist/tests/rm_execution (hunk 2 above)
Comment (martin-g) [reply tests/libs/superlu_dist/tests/rm_execution:20]: Initially (few months ago) this test was using `4`. Then I updated it to `tasks_count 4` which in GHA/Cirrus actually returns `2` (as the number of cores). 
The tests were failing with `2` and passed with `4`. 
Let me try one more time with `tasks_count 4`!

File: tests/libs/superlu_dist/tests/rm_execution (hunk 1 above)
Comment (martin-g) [reply tests/libs/superlu_dist/tests/rm_execution]: Done!

**Comment by martin-g** [comment]
https://github.com/openhpc/ohpc/actions/runs/4314427297/jobs/7527516858 fails with:
```
# Number of processes 2 is smaller than NPROW * NPCOL 4Number of processes 2 is smaller than NPROW * NPCOL 4
```

File: tests/libs/superlu_dist/tests/rm_execution (hunk 2 above)
Comment (martin-g) [reply tests/libs/superlu_dist/tests/rm_execution:20]: With `2` the tests fail with:
`# Number of processes 2 is smaller than NPROW * NPCOL 4Number of processes 2 is smaller than NPROW * NPCOL 4`
